data_loading/dataset.py

```diff
@@ -45,12 +48,40 @@ def __init__(self,
         self._train_test_split = train_test_split
         self._is_train = is_train
         all_files = list_files_with_extension(folder, ".parquet")
-        split_point = int(len(all_files) * train_test_split)
-        if is_train:
-            self._files = all_files[:split_point]
+        if isinstance(train_test_split, (float, int)):
+            split_point = int(len(all_files) * train_test_split)
+            if is_train:
+                self._files = all_files[:split_point]
+            else:
+                self._files = all_files[split_point:]
         else:
-            self._files = all_files[split_point:]
+            if is_train:
+                train_dir = os.path.join(output_folder, "train")
+                if not os.path.exists(train_dir):
+                    os.makedirs(train_dir)
+                out_files = []
+                for file in all_files:
+                    con = duckdb.connect()
+                    con.sql(f"CREATE VIEW sequence AS SELECT * FROM read_parquet('{file}')")
+                    output_path = os.path.join(train_dir, os.path.basename(file))
+                    # get training set and write to temp directory
+                    con.sql(f"COPY(SELECT * FROM sequence where is_training = true) TO '{output_path}' (FORMAT PARQUET)")
+                    out_files.append(output_path)
+                self._files = out_files
+            else:
+                test_dir = os.path.join(output_folder, "test")
+                if not os.path.exists(test_dir):
+                    os.makedirs(test_dir)
+                out_files = []
+                for file in all_files:
+                    con = duckdb.connect()
+                    con.sql(f"CREATE VIEW sequence AS SELECT * FROM read_parquet('{file}')")
+                    output_path = os.path.join(test_dir, os.path.basename(file))
+                    # get test set and write to temp directory
+                    con.sql(f"COPY(SELECT * FROM sequence where is_training = false) TO '{output_path}' (FORMAT PARQUET)")
+                    out_files.append(output_path)
```

Comment:
split person_sequence into training and  validation set preserving partition structure.